File: Kotlin/src/main/kotlin/chapter4_factory/factory_method/PizzaTestDrive.kt
package chapter4_factory.factory_method

object PizzaTestApplication {
    @Throws(Exception::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val nyStore = NYPizzaStore()
        val chicagoSTore = ChicagoPizzaStore()

        val edanPizza = nyStore.orderPizza("cheese")
        println("에단이 주문한 ${edanPizza.name} 완성!\n")

        val zoelPizza = chicagoSTore.orderPizza("cheese")
        println("조엘이 주문한 ${zoelPizza.name} 완성!\n")
    }
}

File: Scala/src/main/scala/factory/PizzaTestDrive.scala
package factory

object PizzaTestDrive extends App {
  val nyStore = new NyPizzaStore()
  val chicagoStore = new ChicagoPizzaStore()

  val nyPizza = nyStore.orderPizza(PizzaType.CHEESE)
  println(s"에단이 주문한 ${nyPizza.getName}")

  val chicagoPizza = chicagoStore.orderPizza(PizzaType.CHEESE)
  println(s"조엘이 주문한 ${chicagoPizza.getName}")
}
